src/python/pants/backend/python/tasks/select_interpreter.py

```diff
@@ -21,6 +21,27 @@
 from pants.util.dirutil import safe_mkdir_for
 
 
+class SelectInterpreterClientMixin(object):
+  """???"""
+
+  def get_selected_interpreter(self):
+    return self.context.products.get_data(PythonInterpreter)
```

Comment:
Could have this mixin request this product.

Reply:
I spent a little while thinking about how to do this, and everything I came up with seemed pretty complex, since this isn't intended to be a base class. If there's an easy way let me know, but given that there are only two mixers right now, I think it might be fine as is? I am actually in support of this change, just can't quite see how to do it as a mixin.

Reply:
"The right thing" happens if you use super in a mixin, afaik.